fs-xattr using hardcoded GitHub runner path

I've installed the macOS build for arm64 from https://github.com/rancher-sandbox/rancher-desktop/actions/runs/16454941484?pr=8933.

When I start it, I get a diagnostic about not being able to find the fs-xattr module:

<img width="2784" height="1664" alt="Image" src="https://github.com/user-attachments/assets/d339a41f-6a6c-4840-bfe3-2646310574d0" />

The path shown is from the GitHub runner used to build the package.

When I press "Rerun" I get a different error:

<img width="2784" height="1664" alt="Image" src="https://github.com/user-attachments/assets/b8b9f2ac-f456-4455-a806-32e08a22a9fb" />

```
2025-07-22T23:52:37.429Z: Failed to import fs-xattr, cannot copy extended attributes from /Users/jan/.bashrc: Error: Cannot find module './build/Release/xattr'
Require stack:
- /Users/runner/work/rancher-desktop/rancher-desktop/node_modules/fs-xattr/index.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1408:15)
    at s._resolveFilename (node:electron/js2c/browser_init:2:129632)
    at defaultResolveImpl (node:internal/modules/cjs/loader:1064:19)
    at resolveForCJSWithHooks (node:internal/modules/cjs/loader:1069:22)
    at Module._load (node:internal/modules/cjs/loader:1218:37)
    at c._load (node:electron/js2c/node_init:2:18013)
    at TracingChannel.traceSync (node:diagnostics_channel:322:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:242:24)
    at Module.require (node:internal/modules/cjs/loader:1494:12)
    at require (node:internal/modules/helpers:135:16) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/runner/work/rancher-desktop/rancher-desktop/node_modules/fs-xattr/index.js'
  ]
}
2025-07-22T23:52:37.431Z: Error: Failed to determine if `/Users/jan/.bashrc` contains extended attributes
2025-07-22T23:53:23.645Z: Error: Failed to copy extended attributes while managing `/Users/jan/.bashrc`
```

--- a/jest.config.js
+++ b/jest.config.js
@@ -12,7 +12,7 @@ export default {
   moduleFileExtensions:    [
     'js',
     'json',
-    'node', // For native modules, e.g. fs-xattr
+    'node', // For native modules, e.g. @napi-rs/xattr
     'ts',
     'vue',
   ],

--- a/package.json
+++ b/package.json
@@ -56,6 +56,7 @@
   "dependencies": {
     "@docker/extension-api-client-types": "0.4.2",
     "@kubernetes/client-node": "0.18.1",
+    "@napi-rs/xattr": "^1.0.2",
     "@rancher/components": "0.3.0-alpha.1",
     "@xterm/addon-fit": "0.10.0",
     "@xterm/addon-search": "0.15.0",
@@ -167,7 +168,6 @@
   },
   "optionalDependencies": {
     "dmg-license": "1.0.11",
-    "fs-xattr": "0.4.0",
     "posix-node": "0.12.0"
   },
   "browserslist": [

--- a/pkg/rancher-desktop/integrations/__tests__/manageLinesInFile.spec.ts
+++ b/pkg/rancher-desktop/integrations/__tests__/manageLinesInFile.spec.ts
@@ -81,10 +81,7 @@ describe('manageLinesInFile', () => {
 
   describe('Target exists as a plain file', () => {
     testUnix('Preserves extended attributes', async() => {
-      // eslint-disable-next-line @typescript-eslint/ban-ts-comment -- This only fails on Windows
-      // @ts-ignore // fs-xattr is not available on Windows.
-      const { listAttributes, getAttribute, setAttribute } = await import('fs-xattr');
-
+      const { listAttributes, getAttribute, setAttribute } = await import('@napi-rs/xattr');
       const unmanagedContents = 'existing lines\n';
       const attributeKey = 'user.io.rancherdesktop.test';
       const attributeValue = 'sample attribute contents';
@@ -113,6 +110,7 @@ describe('manageLinesInFile', () => {
     });
 
     test('Put lines in file that exists and has content', async() => {
+      const { listAttributes } = await import('@napi-rs/xattr');
       const data = 'this is already present in the file\n';
       const expectedContents = [data, START_LINE, TEST_LINE_1, END_LINE, ''].join('\n');
 
@@ -121,9 +119,6 @@ describe('manageLinesInFile', () => {
 
       await expect(fs.promises.readFile(rcFilePath, 'utf8')).resolves.toEqual(expectedContents);
       if (process.platform !== 'win32') {
-        // eslint-disable-next-line @typescript-eslint/ban-ts-comment -- This only fails on Windows
-        // @ts-ignore // fs-xattr is not available on Windows.
-        const { listAttributes } = await import('fs-xattr');
         const allAttrs: string[] = await listAttributes(rcFilePath);
         // filter out attributes like com.apple.provenance that the OS might add
         const filteredAttrs = allAttrs.filter(item => !item.startsWith('com.apple.'));

--- a/pkg/rancher-desktop/integrations/manageLinesInFile.ts
+++ b/pkg/rancher-desktop/integrations/manageLinesInFile.ts
@@ -152,15 +152,16 @@ export default async function manageLinesInFile(path: string, desiredManagedLine
  * files must already exist.
  */
 async function copyFileExtendedAttributes(fromPath: string, toPath: string): Promise<void> {
+  const { listAttributes, getAttribute, removeAttribute, setAttribute } = await import('@napi-rs/xattr');
   try {
-    // eslint-disable-next-line @typescript-eslint/ban-ts-comment -- This only fails on Windows
-    // @ts-ignore // fs-xattr is not available on Windows
-    const { listAttributes, getAttribute, setAttribute } = await import('fs-xattr');
-
     for (const attr of await listAttributes(fromPath)) {
       const value = await getAttribute(fromPath, attr);
 
-      await setAttribute(toPath, attr, value);
+      if (value === null) {
+        await removeAttribute(toPath, attr);
+      } else {
+        await setAttribute(toPath, attr, value);
+      }
     }
   } catch (cause) {
     if (process.env.NODE_ENV === 'test' && process.env.RD_TEST !== 'e2e') {

